commands/project_sync.go

```diff
@@ -86,35 +86,228 @@ func (cmd *ProjectSync) Commands() []cli.Command {
+func (cmd *ProjectSync) RunCheck(ctx *cli.Context) error {
+	cmd.out.Spin("Preparing test of unison filesync...")
+	volumeName, workingDir, err := cmd.initializeSettings(ctx.String("dir"))
+	if err != nil {
+		return cmd.Failure(err.Error(), "SYNC-PATH-ERROR", 12)
+	}
+	cmd.out.Info("Ready to begin unison test")
+	cmd.out.Spin("Checking for unison container...")
+	if running := util.ContainerRunning(volumeName); !running {
+		cmd.out.Error("Unison container (%s) is not running", volumeName)
+		return cmd.Failure(err.Error(), "SYNC-CHECK-FAILED", 13)
+	}
+	cmd.out.Info("Unison container found: %s", volumeName)
+	cmd.out.Spin("Check unison container process is listening...")
+	if _, err := cmd.WaitForUnisonContainer(volumeName, ctx.Int("initial-sync-timeout")); err != nil {
+		cmd.out.Error("Unison process not listening")
+		return cmd.Failure(err.Error(), "SYNC-CHECK-FAILED", 13)
+	}
+	cmd.out.Info("Unison process is listening")
+
+	// Determine if sync progress can be tracked.
+	//cmd.out.Spin("Syncing a test file...")
+	cmd.out.Info("Preparing live file sync test")
+	var logFile = cmd.LogFileName(volumeName)
+	if err := cmd.WaitForSyncInit(logFile, workingDir, ctx.Int("initial-sync-timeout"), ctx.Int("initial-sync-wait")); err != nil {
+		return cmd.Failure(err.Error(), "UNISON-SYNC-FAILED", 13)
+	}
+
+	// Sidestepping the notification so rig sync:check can be run as a background process.
+	cmd.out.Info("Sync check completed successfully")
+	return nil
+}
+
+// RunPurge cleans out the project sync state.
+func (cmd *ProjectSync) RunPurge(ctx *cli.Context) error {
+	if util.IsLinux() {
+		return cmd.Success("No Unison process to clean up.")
+	}
+
+	volumeName, workingDir, err := cmd.initializeSettings(ctx.String("dir"))
+	if err != nil {
+		return cmd.Failure(err.Error(), "SYNC-PATH-ERROR", 12)
+	}
+
+	cmd.out.Spin("Checking for unison container...")
+	if running := util.ContainerRunning(volumeName); running {
+		cmd.out.Spin(fmt.Sprintf("Stopping Unison container (%s)", volumeName))
+		if err := util.Command("docker", "container", "stop", volumeName).Run(); err != nil {
+			cmd.out.Warn("Could not stop unison container (%s): Maybe it's already stopped?", volumeName)
+		} else {
+			cmd.out.Info("Stopped unison container (%s)", volumeName)
+		}
+	} else {
+		cmd.out.Info("No running unison container.")
+	}
+
+	logFile := cmd.LogFileName(volumeName)
+	cmd.out.Spin(fmt.Sprintf("Removing unison log file: %s", logFile))
+	if util.FileExists(logFile, workingDir) {
+		if err := util.RemoveFile(logFile, workingDir); err != nil {
+			cmd.out.Error("Could not remove unison log file: %s: %s", logFile, err.Error())
+		} else {
+			cmd.out.Info("Removed unison log file: %s", logFile)
+		}
+	} else {
+		cmd.out.Info("Log file does not exist")
+	}
+
+	// Remove sync fragment files.
+	cmd.out.Spin("Removing .unison directories")
+	if err := util.RemoveFileGlob("*.unison*", workingDir, cmd.out); err != nil {
```

Comment:
I don't have a lot of experience with these fragment directories. Do both the directory names and all the files within them match the *.unison* pattern (which I believe they would need to in order for this cleanup to succeed).

Reply:
I tested this with various files, here's an example:

`.unison.._.DS_Store.94df3249bac7c3ee51c590d510133947.unison.tmp`

I could not remember if I'd see something prefixing `.unison` so went ahead to cover both.